examples/md-flexible/scripts/predict.py

```diff
@@ -0,0 +1,112 @@
+
+    Returns:
+        models (dict): A dictionary containing the trained models for each target variable.
+        label_encoders (dict): A dictionary containing LabelEncoders for decoding the predictions.
+    """
+    # Throw an error if the model file does not exist
+    if not os.path.exists(model_file):
+        raise FileNotFoundError(f"Model file not found: {model_file}")
+
+    # Load the models and label encoders from the pickle file
+    with open(model_file, 'rb') as f:
+        combined_data = pickle.load(f)
+    # Return the models and label encoders
+    return combined_data['models'], combined_data['label_encoders']
+
+
+def preprocess_live_info(live_info: dict) -> pd.DataFrame:
+    """
+    Preprocess the live info received from C++ into a format suitable for model input.
+
+    This function selects the relevant features from the live info dictionary (received from C++),
+    formats them into a pandas DataFrame, and prepares them for input into the trained models.
+
+    Args:
+        live_info (dict): A dictionary containing the live info features from C++.
+
+    Returns:
+        pd.DataFrame: A DataFrame containing the processed features ready for prediction.
+    """
+    # Define the required features for the model input
+    features = ['avgParticlesPerCell', 'maxParticlesPerCell', 'homogeneity', 'maxDensity', 'particlesPerCellStdDev',
+                'threadCount']
+
+    # Extract the relevant features from live info
+    live_info_input = {feature: live_info[feature] for feature in features}
+
+    # Convert the input into a pandas DataFrame with one row
+    live_info_df = pd.DataFrame([live_info_input])
+
+    return live_info_df
+
+
+def predict(live_info: dict, models: dict, label_encoders: dict) -> dict:
+    """
+    Perform a forward pass through the trained models to predict the tuning configuration.
+
+    This function takes the live info data, preprocesses it, and makes predictions using the loaded models.
+    The predictions are then decoded back into their original labels using the corresponding LabelEncoders.
+
+    Args:
+        live_info (dict): A dictionary of live info data from C++.
+        models (dict): A dictionary containing the trained models for each target variable.
+        label_encoders (dict): A dictionary containing LabelEncoders for decoding predictions.
+
+    Returns:
+        dict: A dictionary containing the predicted tuning configuration (Container, Traversal, etc.).
+    """
+    live_info_df = preprocess_live_info(live_info)
+
+    predictions = {}
+
+    # Perform prediction for each target
+    for target, model in models.items():
+        prediction_encoded = model.predict(live_info_df)
+        prediction = label_encoders[target].inverse_transform(prediction_encoded)
+        predictions[target] = prediction[0]
+
+    # Convert CellSizeFactor to string as a workaround as sending float values caused issues
```

Comment:
Returning a value of type double caused an issue when assigning it to config.CellSizeFactor. Since I changed to only tuning `container`, `traversal`, `dataLayout` and `newton3`, this line will also be changed.